Services/Dashboard/Block/classes/class.ilDashboardBlockGUI.php

```diff
@@ -0,0 +1,782 @@
+                    $items[] = $item;
+                }
+            }
+            $groupedCards[] = $this->factory->item()->group((string) $title, $items);
+        }
+
+
+        return $groupedCards;
+    }
+
+
+    protected function getListItemForData(array $data): ?\ILIAS\UI\Component\Item\Item
+    {
+        $itemListGui = $this->byType($data['type']);
+        $this->addCustomCommandsToActionMenu($itemListGui, $data['ref_id']);
+        ilObjectActivation::addListGUIActivationProperty($itemListGui, $data);
+
+        $list_item = $itemListGui->getAsListItem(
+            (int) $data['ref_id'],
+            (int) $data['obj_id'],
+            (string) $data['type'],
+            (string) $data['title'],
+            (string) $data['description']
+        );
+
+        return $list_item;
+    }
+
+    protected function isRepositoryObject(): bool
+    {
+        return false;
+    }
+
+    protected function getLegacyContent(): string
+    {
+        $groupedCards = [];
+        foreach ($this->loadData() as $title => $group) {
+            $cards = [];
+            foreach ($group as $datum) {
+                $cards[] = $this->getCardForData($datum);
+            }
+            $cards = array_filter($cards);
+            if ($cards) {
+                $groupedCards[] = $this->ui->factory()->panel()->sub(
+                    $title,
+                    $this->factory->deck($cards)->withNormalCardsSize()
+                );
+            }
+        }
+
+        if ($groupedCards) {
+            return $this->renderer->render($groupedCards);
+        }
+
+        return $this->getNoItemFoundContent();
+    }
+
+    public function getNoItemFoundContent(): string
+    {
+        return $this->emptyHandling();
+    }
+
+    public function getViewSettings(): ilPDSelectedItemsBlockViewSettings
+    {
+        return $this->viewSettings;
+    }
+
+    protected function initAndShow(): void
+    {
+        $this->initViewSettings();
+        $this->viewSettings->parse();
+        $this->initData();
+
+        if ($this->viewSettings->isTilePresentation()) {
+            $this->setPresentation(self::PRES_MAIN_LEG);
+        } else {
+            $this->setPresentation(self::PRES_MAIN_LIST);
+        }
+
+        if ($this->ctrl->isAsynch()) {
+            echo $this->getHTML();
+            exit;
+        }
+
+        $this->returnToContext();
+    }
+
+    public function getHTML(): string
+    {
+        $this->setTitle(
+            $this->lng->txt('dash_' . $this->viewSettings->getViewName($this->viewSettings->getCurrentView()))
+        );
+
+        if (!$this->data) {
+            return $this->emptyHandling();
+        }
+
+        $this->addCommandActions();
+        $this->setData($this->getItemGroups());
+
+        return parent::getHTML();
+    }
+
+    public function setData(array $a_data): void
+    {
+        $this->data = array_filter($a_data);
+    }
+
+    public function groupItemsByStartDate(): array
```

Comment:
Add a default return decription inside the docs. Either something like `<string, int>` or, if this is not definable, a descriptive sentence.

Same for other array return or parameters